test/oli_web/live/delivery/student_dashboard/course_content_live_test.exs

```diff
@@ -0,0 +1,419 @@
+defmodule OliWeb.Delivery.StudentDashboard.CourseContentLiveTest do
+  use ExUnit.Case, async: true
+  use OliWeb.ConnCase
+
+  import Phoenix.LiveViewTest
+  import Ecto.Query
+
+  alias OliWeb.Delivery.StudentDashboard.CourseContentLive
+  alias Lti_1p3.Tool.ContextRoles
+  alias Oli.Delivery.Sections
+  alias Oli.Delivery.Attempts.Core
+  alias Oli.Delivery.Sections.SectionResource
+  alias Oli.Repo
+  alias Oli.Delivery.Gating
+
+  defp isolated_live_view_course_content(conn, section_slug, user_id) do
+    live_isolated(conn, CourseContentLive,
+      session: %{"section_slug" => section_slug, "current_user_id" => user_id}
+    )
+  end
+
+  defp section_with_larger_hierarchy(%{user: user} = _conn) do
+    %{section: section, mod1_pages: mod1_pages} = Oli.Seeder.base_project_with_larger_hierarchy()
+    Sections.enroll(user.id, section.id, [ContextRoles.get_role(:context_learner)])
+
+    %{section: section, mod1_pages: mod1_pages}
+  end
+
+  describe "course renders the browser correctly" do
+    setup [:user_conn, :section_with_larger_hierarchy]
+
+    test "with left arrow disabled", %{conn: conn, user: user, section: section} do
+      {:ok, view, _html} = isolated_live_view_course_content(conn, section.slug, user.id)
+
+      assert has_element?(
+               view,
+               "button[phx-click=\"previous_node\"].grayscale.pointer-events-none"
+             )
+    end
+
+    test "with breadcrumbs initially set to Curriculum", %{
+      conn: conn,
+      user: user,
+      section: section
+    } do
+      {:ok, view, _html} = isolated_live_view_course_content(conn, section.slug, user.id)
+
+      assert breadcrumbs_length(view) == 1
+      assert has_element?(view, "button[phx-click=\"breadcrumb-navigate\"]", "Curriculum")
+    end
+
+    test "navigates left and right", %{conn: conn, user: user, section: section} do
```

Comment:
Same as above, but with the units and pages that are being created in section_with_larger_hierarchy